Release/2.1.5 (#5305)
nautobot/nautobot: nautobot/docs/release-notes/version-2.1.md

- [#5243](https://github.com/nautobot/nautobot/issues/5243) - Added support for setting display_field on DynamicModelChoiceField to nested values in suggested choices list.

### Changed

- [#5171](https://github.com/nautobot/nautobot/issues/5171) - Changed the tagging of `nautobot-dev` Docker images to reserve the `latest` and `latest-py<version>` tags for the latest stable release of Nautobot, rather than the latest build from the `develop` branch.
- [#5254](https://github.com/nautobot/nautobot/issues/5254) - Changed `TreeQuerySet.ancestors` implementation to a more efficient approach for shallow trees.
- [#5254](https://github.com/nautobot/nautobot/issues/5254) - Changed the location detail view not to annotate tree fields on its queries.
- [#5267](https://github.com/nautobot/nautobot/issues/5267) - Updated navbar user dropdown with chevron.
- [#5267](https://github.com/nautobot/nautobot/issues/5267) - Reorganized navbar css rules in `base.css`.

### Fixed

- [#5058](https://github.com/nautobot/nautobot/issues/5058) - Changed more filter parameters from location_id to location in virtualization/forms.py

> - [#5121](https://github.com/nautobot/nautobot/issues/5121) - Fix deleting git repository results in creation of job result stuck in running state.

Reviewer: ```suggestion
- [#5121](https://github.com/nautobot/nautobot/issues/5121) - Fixed an issue where deleting a git repository resulted in a job result stuck in running state.
```

- [#5267](https://github.com/nautobot/nautobot/issues/5267) - Updated navbar user dropdown with chevron.
- [#5267](https://github.com/nautobot/nautobot/issues/5267) - Reorganized navbar css rules in `base.css`.

### Security

- [#5303](https://github.com/nautobot/nautobot/pull/5303) - Updated `cryptography` to 42.0.2 due to CVE-2024-0727. This is not a direct dependency so will not auto-update when upgrading. Please be sure to upgrade your local environment.

> 

Reviewer: `Security` should come before `Added` in our usual ordering of release-notes.
